File: service/src/main/java/com/iberianmotorsports/service/service/GridRaceService.java
package com.iberianmotorsports.service.service;

import com.iberianmotorsports.service.model.Grid;
import com.iberianmotorsports.service.model.GridRace;
import com.iberianmotorsports.service.model.Race;

import java.util.List;

public interface GridRaceService {

    GridRace getGridRace(Long gridId, Long raceId);

    List<GridRace> getGridRaceForRace(Long raceId);

    void calculateDropRoundForGrid(Grid grid);

    void calculateDropRoundForRaceChampionship(Race race);

    void calculateGridRace(Long raceId);

    GridRace saveGridRace(GridRace gridRace);

    void calculatePointsToGridRace(GridRace gridRace, int position, GridRace winner);

}

File: frontend/src/app/service/gridrace.service.ts
import {Injectable} from "@angular/core";
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {GridRace} from "../model/GridRace";

@Injectable()
export class GridRaceService {
    private url: string= environment.apiPath + 'grid-race';

    constructor(
        private httpClient: HttpClient
    ) {
    }

    getGridRace(gridId: number, raceId: number): Observable<GridRace> {
        const url = this.url + "/" + gridId + "/" + raceId;
        return this.httpClient.get<GridRace>(url);
    }

    getGridRaceForRace(raceId: number): Observable<GridRace[]> {
        const url = this.url + "/public/" + raceId;
        return this.httpClient.get<GridRace[]>(url);
    }

}
